In [ ]:
import pandas as pd
import numpy as np
import plotly.express as px
import plotly.graph_objects as go
import warnings
warnings.filterwarnings('ignore')

#Key Questions:-



1.Find Out The Co2_Emissions_MetricTons.

2.Find Out The Energy_Consumption_TWh.

3.Find Out The Automobile_Co2_Emissions_MetricTons.

4.Find Out The Industrial_Co2_Emissions_MetricTons.

5.Find Out The Agriculture_Co2_Emissions_MetricTons.

6.Find Out The Domestic_Co2_Emissions_MetricTons.

7.Find Out The Population.

8.Find Out The GDP_Billion_USD.

9.Find Out The Urbanization_Percentage.

10.Find Out The Renewable_Energy_Percentage.

11.Find Out The Industrial_Growth_Percentage.

12.Find Out The Transport_Growth_Percentage.




In [ ]:
df=pd.read_csv('/content/drive/MyDrive/Data/co2 Project.csv')

In [ ]:
df

Industry_Type         Region       Country  Year  \
0       Construction  North America        Brazil  2010   
1             Mining         Europe       Germany  2006   
2      Manufacturing  South America  South Africa  2017   
3       Construction         Europe         India  2018   
4       Construction         Africa         China  2013   
...              ...            ...           ...   ...   
17681  Manufacturing           Asia  South Africa  2020   
17682         Energy         Africa       Germany  2009   
17683         Mining  South America        Canada  2017   
17684         Energy         Europe           USA  2012   
17685  Manufacturing        Oceania         China  2004   

       Co2_Emissions_MetricTons  Energy_Consumption_TWh  \
0                         89.13                   90.14   
1                        224.81                  931.70   
2                        179.72                  255.14   
3                         23.28                  887.29   
4                        124.54                  923.03   
...                         ...                     ...   
17681                    177.73                  132.27   
17682                     11.89                  615.87   
17683                     47.94                  564.24   
17684                    211.15                  826.48   
17685                    200.73                  970.36   

       Automobile_Co2_Emissions_MetricTons  \
0                                    98.37   
1                                    10.81   
2                                    55.36   
3                                    78.99   
4                                    65.93   
...                                    ...   
17681                                32.44   
17682                                71.90   
17683                                32.63   
17684                                36.02   
17685                                55.00   

       Industrial_Co2_Emissions_MetricTons  \
0                                   118.36   
1                                    66.73   
2                                   111.67   
3                                   123.60   
4                                    52.32   
...                                    ...   
17681                                90.00   
17682                               122.81   
17683                                47.54   
17684                               122.83   
17685                                36.85   

       Agriculture_Co2_Emissions_MetricTons  \
0                                     31.41   
1                                     39.45   
2                                      1.25   
3                                     46.81   
4                                     35.67   
...                                     ...   
17681                                 49.83   
17682                                  3.05   
17683                                 47.54   
17684                                 19.24   
17685                                 27.19   

       Domestic_Co2_Emissions_MetricTons  Population_Millions  \
0                                   0.77               941.06   
1                                   0.21              1421.94   
2                                   4.97               522.75   
3                                  13.77              1304.60   
4                                  13.91              1437.68   
...                                  ...                  ...   
17681                              10.25               849.67   
17682                               1.50               581.74   
17683                              15.02               898.74   
17684                              15.82                28.58   
17685                               5.70               591.45   

       GDP_Billion_USD  Urbanization_Percentage  Renewable_Energy_Percentage  \
0             13096.32                    52.79          

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17686 entries, 0 to 17685
Data columns (total 16 columns):
 #   Column                                Non-Null Count  Dtype  
---  ------                                --------------  -----  
 0   Industry_Type                         17686 non-null  object 
 1   Region                                17686 non-null  object 
 2   Country                               17686 non-null  object 
 3   Year                                  17686 non-null  int64  
 4   Co2_Emissions_MetricTons              17686 non-null  float64
 5   Energy_Consumption_TWh                17686 non-null  float64
 6   Automobile_Co2_Emissions_MetricTons   17686 non-null  float64
 7   Industrial_Co2_Emissions_MetricTons   17686 non-null  float64
 8   Agriculture_Co2_Emissions_MetricTons  17686 non-null  float64
 9   Domestic_Co2_Emissions_MetricTons     17686 non-null  float64
 10  Population_Millions                   17686 non-null  float64
 11  GDP_Billion_USD

In [ ]:
df.describe()

Year  Co2_Emissions_MetricTons  Energy_Consumption_TWh  \
count  17686.000000              17686.000000            17686.000000   
mean    2011.024935                150.331667              500.071501   
std        6.657483                 86.401588              287.008603   
min     2000.000000                  0.500000                0.630000   
25%     2005.000000                 75.580000              252.447500   
50%     2011.000000                150.820000              499.645000   
75%     2017.000000                225.247500              750.065000   
max     2022.000000                299.990000              999.880000   

       Automobile_Co2_Emissions_MetricTons  \
count                         17686.000000   
mean                             49.983267   
std                              28.936346   
min                               0.110000   
25%                              24.712500   
50%                              50.305000   
75%                              75.200000   
max                             100.000000   

       Industrial_Co2_Emissions_MetricTons  \
count                         17686.000000   
mean                             99.975617   
std                              57.630602   
min                               0.110000   
25%                              49.550000   
50%                             100.385000   
75%                             149.867500   
max                             200.000000   

       Agriculture_Co2_Emissions_MetricTons  \
count                          17686.000000   
mean                              24.895109   
std                               14.422553   
min                                0.100000   
25%                               12.390000   
50%                               24.820000   
75%                               37.310000   
max                               50.000000   

       Domestic_Co2_Emissions_MetricTons  Population_Millions  \
count                       17686.000000         17686.000000   
mean                           10.170034           752.966182   
std                             5.757755           432.819846   
min                             0.100000             0.510000   
25%                             5.160000           377.497500   
50%                            10.200000           750.400000   
75%                            15.190000          1126.875000   
max                            20.000000          1499.830000   

       GDP_Billion_USD  Urbanization_Percentage  Renewable_Energy_Percentage  \
count     17686.000000             17686.000000                 17686.000000   
mean      12522.789595                60.039285                    49.916005   
std        7184.289404                23.131004                    28.962171   
min           0.220000                20.010000                     0.000000   
25%        6392.367500                39.802500                    24.730000   
50%       12491.915000                60.260000                    50.000000   
75%       18664.797500                80.030000                    75.057500   
max       24999.570000               100.000000                   100.000000   

       Industrial_Growth_Percentage  Transport_Growth_Percentage  
count                  17686.000000                 17686.000000  
mean                       2.570492                     2.471663  
std                        7.229444                     4.338840  
min                      -10.000000                    -5.000000  
25%                       -3.727500                    -1.280000  
50%                        2.500000                     2.470000  
75%                        8.770000                     6.230000  
max                       15.000000                    10.000000

In [ ]:
df.columns

Index(['Industry_Type', 'Region', 'Country', 'Year',
       'Co2_Emissions_MetricTons', 'Energy_Consumption_TWh',
       'Automobile_Co2_Emissions_MetricTons',
       'Industrial_Co2_Emissions_MetricTons',
       'Agriculture_Co2_Emissions_MetricTons',
       'Domestic_Co2_Emissions_MetricTons', 'Population_Millions',
       'GDP_Billion_USD', 'Urbanization_Percentage',
       'Renewable_Energy_Percentage', 'Industrial_Growth_Percentage',
       'Transport_Growth_Percentage'],
      dtype='object')

In [ ]:
df.shape

(17686, 16)

In [ ]:
df.isnull().sum()

Industry_Type                           0
Region                                  0
Country                                 0
Year                                    0
Co2_Emissions_MetricTons                0
Energy_Consumption_TWh                  0
Automobile_Co2_Emissions_MetricTons     0
Industrial_Co2_Emissions_MetricTons     0
Agriculture_Co2_Emissions_MetricTons    0
Domestic_Co2_Emissions_MetricTons       0
Population_Millions                     0
GDP_Billion_USD                         0
Urbanization_Percentage                 0
Renewable_Energy_Percentage             0
Industrial_Growth_Percentage            0
Transport_Growth_Percentage             0
dtype: int64

In [ ]:
df.head()

Industry_Type         Region       Country  Year  Co2_Emissions_MetricTons  \
0   Construction  North America        Brazil  2010                     89.13   
1         Mining         Europe       Germany  2006                    224.81   
2  Manufacturing  South America  South Africa  2017                    179.72   
3   Construction         Europe         India  2018                     23.28   
4   Construction         Africa         China  2013                    124.54   

   Energy_Consumption_TWh  Automobile_Co2_Emissions_MetricTons  \
0                   90.14                                98.37   
1                  931.70                                10.81   
2                  255.14                                55.36   
3                  887.29                                78.99   
4                  923.03                                65.93   

   Industrial_Co2_Emissions_MetricTons  Agriculture_Co2_Emissions_MetricTons  \
0                               118.36                                 31.41   
1                                66.73                                 39.45   
2                               111.67                                  1.25   
3                               123.60                                 46.81   
4                                52.32                                 35.67   

   Domestic_Co2_Emissions_MetricTons  Population_Millions  GDP_Billion_USD  \
0                               0.77               941.06         13096.32   
1                               0.21              1421.94         24337.71   
2                               4.97               522.75         24523.17   
3                              13.77              1304.60         12616.18   
4                              13.91              1437.68          4476.00   

   Urbanization_Percentage  Renewable_Energy_Percentage  \
0                    52.79                         7.78   
1                    50.18                        31.52   
2                    65.22                         5.91   
3                    23.69                         7.52   
4                    94.64                         8.54   

   Industrial_Growth_Percentage  Transport_Growth_Percentage  
0                         11.17                         2.93  
1                         13.34                         9.30  
2                         -9.88                         4.77  
3                         -0.64                         8.21  
4                          5.98                         0.84

#Types Of Industry Distribution.

In [ ]:
industry_counts = df['Industry_Type'].value_counts()

In [ ]:
fig = px.pie(industry_counts,
             values=industry_counts.values,
             names=industry_counts.index,
             title='Industry Type Distribution',
             hole=0.3,
             color_discrete_sequence=px.colors.qualitative.Pastel)

fig.show()

#Industry Type Distribution By Region

In [ ]:
region_industry_counts = df.groupby(['Region', 'Industry_Type'])['Industry_Type'].count().reset_index(name='Count')

In [ ]:
fig = px.sunburst(region_industry_counts, path=['Region', 'Industry_Type'], values='Count',
                  title='Industry Type Distribution by Region')

fig.update_traces(textinfo="label+percent parent")
fig.show()

#1.Find Out The Co2_Emissions_MetricTons.

In [ ]:
co2_emissions_by_factors = df.groupby(['Region', 'Country', 'Year'])['Co2_Emissions_MetricTons'].sum().reset_index()

In [ ]:
fig = px.bar(co2_emissions_by_factors,
             x='Year',
             y='Co2_Emissions_MetricTons',
             color='Country',
             facet_col='Region',
             title='CO2 Emissions by Region, Country, and Year')

fig.show()

In [ ]:
country_region_date_emissions = df.groupby(['Industry_Type'])['Co2_Emissions_MetricTons'].sum().reset_index()

In [ ]:
fig = px.line(country_region_date_emissions,
              x='Industry_Type',
              y='Co2_Emissions_MetricTons',
              title='CO2 Emissions by Industry Type')
fig.show()

#2.Find Out The Energy_Consumption_TWh

In [ ]:
energy_consumption_by_factors = df.groupby(['Region', 'Country', 'Year'])['Energy_Consumption_TWh'].sum().reset_index()

In [ ]:
fig = px.bar(energy_consumption_by_factors,
             x='Year',
             y='Energy_Consumption_TWh',
             color='Country',
             facet_col='Region',
             title='Energy Consumption by Region, Country, and Year')

fig.show()

In [ ]:
energy_consumption_by_factors = df.groupby(['Industry_Type'])['Energy_Consumption_TWh'].sum().reset_index()

In [ ]:
fig = px.pie(energy_consumption_by_factors,
             values='Energy_Consumption_TWh',
             names='Industry_Type',
             title='Energy Consumption by Industry Type',
             hole=0.4,  # Create a hole for the donut chart
             color_discrete_sequence=px.colors.qualitative.Pastel)

fig.show()

In [ ]:
df.head()

Industry_Type         Region       Country  Year  Co2_Emissions_MetricTons  \
0   Construction  North America        Brazil  2010                     89.13   
1         Mining         Europe       Germany  2006                    224.81   
2  Manufacturing  South America  South Africa  2017                    179.72   
3   Construction         Europe         India  2018                     23.28   
4   Construction         Africa         China  2013                    124.54   

   Energy_Consumption_TWh  Automobile_Co2_Emissions_MetricTons  \
0                   90.14                                98.37   
1                  931.70                                10.81   
2                  255.14                                55.36   
3                  887.29                                78.99   
4                  923.03                                65.93   

   Industrial_Co2_Emissions_MetricTons  Agriculture_Co2_Emissions_MetricTons  \
0                               118.36                                 31.41   
1                                66.73                                 39.45   
2                               111.67                                  1.25   
3                               123.60                                 46.81   
4                                52.32                                 35.67   

   Domestic_Co2_Emissions_MetricTons  Population_Millions  GDP_Billion_USD  \
0                               0.77               941.06         13096.32   
1                               0.21              1421.94         24337.71   
2                               4.97               522.75         24523.17   
3                              13.77              1304.60         12616.18   
4                              13.91              1437.68          4476.00   

   Urbanization_Percentage  Renewable_Energy_Percentage  \
0                    52.79                         7.78   
1                    50.18                        31.52   
2                    65.22                         5.91   
3                    23.69                         7.52   
4                    94.64                         8.54   

   Industrial_Growth_Percentage  Transport_Growth_Percentage  
0                         11.17                         2.93  
1                         13.34                         9.30  
2                         -9.88                         4.77  
3                         -0.64                         8.21  
4                          5.98                         0.84

#3.Find Out The Automobile_Co2_Emissions_MetricTons

In [ ]:
automobile_emissions_by_factors = df.groupby(['Region', 'Country', 'Year'])['Automobile_Co2_Emissions_MetricTons'].sum().reset_index()

In [ ]:
fig = px.bar(automobile_emissions_by_factors,
             x='Year',
             y='Automobile_Co2_Emissions_MetricTons',
             color='Country',
             facet_col='Region',
             title='Automobile CO2 Emissions by Region, Country, and Year')

fig.show()

In [ ]:
automobile_emissions_by_factors = df.groupby(['Industry_Type'])['Automobile_Co2_Emissions_MetricTons'].sum().reset_index()

In [ ]:
fig = px.line(automobile_emissions_by_factors,
              x='Industry_Type',
              y='Automobile_Co2_Emissions_MetricTons',
              title='Automobile CO2 Emissions by Industry Type')
fig.show()

#4.Find Out The Industrial_Co2_Emissions_MetricTons.

In [ ]:
industrial_emissions_by_factors = df.groupby(['Region', 'Country', 'Year'])['Industrial_Co2_Emissions_MetricTons'].sum().reset_index()

In [ ]:
fig = px.area(industrial_emissions_by_factors,
             x='Year',
             y='Industrial_Co2_Emissions_MetricTons',
             color='Country',
             facet_col='Region',
             title='Industrial CO2 Emissions by Region, Country, and Year')

fig.show()

In [ ]:
industrial_emissions_by_factors = df.groupby(['Industry_Type'])['Industrial_Co2_Emissions_MetricTons'].sum().reset_index()

In [ ]:
fig = px.scatter(industrial_emissions_by_factors,
                 x='Industry_Type',
                 y='Industrial_Co2_Emissions_MetricTons',
                 size='Industrial_Co2_Emissions_MetricTons',
                 title='Industrial CO2 Emissions by Industry Type (Bubble Chart)')
fig.show()

#5.Find Out The Agriculture_Co2_Emissions_MetricTons

In [ ]:
agriculture_emissions_by_factors = df.groupby(['Region', 'Country', 'Year'])['Agriculture_Co2_Emissions_MetricTons'].sum().reset_index()

In [ ]:
fig = px.line(agriculture_emissions_by_factors,
              x='Year',
              y='Agriculture_Co2_Emissions_MetricTons',
              color='Country',
              facet_col='Region',
              title='Agriculture CO2 Emissions by Region, Country, and Year')

fig.show()

In [ ]:
agriculture_emissions_by_factors = df.groupby(['Industry_Type'])['Agriculture_Co2_Emissions_MetricTons'].sum().reset_index()

In [ ]:
fig = px.scatter(agriculture_emissions_by_factors,
                 x='Industry_Type',
                 y='Agriculture_Co2_Emissions_MetricTons',
                 size='Agriculture_Co2_Emissions_MetricTons',
                 title='Agriculture CO2 Emissions by Industry Type (Bubble Chart)',
                 color='Industry_Type',
                 hover_name='Industry_Type',  # Display Industry_Type on hover
                 labels={'Agriculture_Co2_Emissions_MetricTons': 'Agriculture CO2 Emissions'},
                 size_max=30
                )

fig.show()

In [ ]:
df.head()

Industry_Type         Region       Country  Year  Co2_Emissions_MetricTons  \
0   Construction  North America        Brazil  2010                     89.13   
1         Mining         Europe       Germany  2006                    224.81   
2  Manufacturing  South America  South Africa  2017                    179.72   
3   Construction         Europe         India  2018                     23.28   
4   Construction         Africa         China  2013                    124.54   

   Energy_Consumption_TWh  Automobile_Co2_Emissions_MetricTons  \
0                   90.14                                98.37   
1                  931.70                                10.81   
2                  255.14                                55.36   
3                  887.29                                78.99   
4                  923.03                                65.93   

   Industrial_Co2_Emissions_MetricTons  Agriculture_Co2_Emissions_MetricTons  \
0                               118.36                                 31.41   
1                                66.73                                 39.45   
2                               111.67                                  1.25   
3                               123.60                                 46.81   
4                                52.32                                 35.67   

   Domestic_Co2_Emissions_MetricTons  Population_Millions  GDP_Billion_USD  \
0                               0.77               941.06         13096.32   
1                               0.21              1421.94         24337.71   
2                               4.97               522.75         24523.17   
3                              13.77              1304.60         12616.18   
4                              13.91              1437.68          4476.00   

   Urbanization_Percentage  Renewable_Energy_Percentage  \
0                    52.79                         7.78   
1                    50.18                        31.52   
2                    65.22                         5.91   
3                    23.69                         7.52   
4                    94.64                         8.54   

   Industrial_Growth_Percentage  Transport_Growth_Percentage  
0                         11.17                         2.93  
1                         13.34                         9.30  
2                         -9.88                         4.77  
3                         -0.64                         8.21  
4                          5.98                         0.84

#6.Find Out The Domestic_Co2_Emissions_MetricTons.

In [ ]:
domestic_co2_emissions_by_factors = df.groupby(['Region', 'Country', 'Year'])['Domestic_Co2_Emissions_MetricTons'].sum().reset_index()

In [ ]:
fig = px.area(domestic_co2_emissions_by_factors,
             x='Year',
             y='Domestic_Co2_Emissions_MetricTons',
             color='Country',
             facet_col='Region',
             title='Domestic CO2 Emissions by Region, Country, and Year')

fig.show()

In [ ]:
domestic_co2_emissions_by_factors = df.groupby(['Industry_Type'])['Domestic_Co2_Emissions_MetricTons'].sum().reset_index()

In [ ]:
fig = px.bar(domestic_co2_emissions_by_factors,
             x='Industry_Type',
             y='Domestic_Co2_Emissions_MetricTons',
             title='Domestic CO2 Emissions by Industry Type')

fig.show()

#7.Find Out The Population.

In [2]:
population_by_factors = df.groupby(['Region', 'Country', 'Year'])['Population_Millions'].sum().reset_index()

In [ ]:
fig = px.bar(population_by_factors,
             x='Year',
             y='Population_Millions',
             color='Country',
             facet_col='Region',
             title='Population by Region, Country, and Year')

fig.show()

In [ ]:
population_by_factors = df.groupby(['Industry_Type'])['Population_Millions'].sum().reset_index()

In [ ]:
fig = px.pie(population_by_factors,
             values='Population_Millions',
             names='Industry_Type',
             title='Population by Industry Type',
             hole=0.4,
             color_discrete_sequence=px.colors.qualitative.Pastel)

fig.show()

#8.Find Out The GDP_Billion_USD.

In [ ]:
gdp_by_factors = df.groupby(['Region', 'Country', 'Year'])['GDP_Billion_USD'].sum().reset_index()

In [ ]:
fig = px.bar(gdp_by_factors,
             x='Year',
             y='GDP_Billion_USD',
             color='Country',
             facet_col='Region',
             title='GDP by Region, Country, and Year')

fig.show()

In [ ]:
gdp_by_factors = df.groupby(['Industry_Type'])['GDP_Billion_USD'].sum().reset_index()

In [ ]:
fig = px.pie(gdp_by_factors,
             values='GDP_Billion_USD',
             names='Industry_Type',
             title='GDP by Industry Type (Donut Chart)',
             hole=0.4)

fig.show()

#9.Find Out The Urbanization_Percentage.

In [ ]:
urbanization_by_factors = df.groupby(['Region', 'Country', 'Year'])['Urbanization_Percentage'].sum().reset_index()

fig = px.bar(urbanization_by_factors,
             x='Year',
             y='Urbanization_Percentage',
             color='Country',
             facet_col='Region',
             title='Urbanization Percentage by Region, Country, and Year')

fig.show()

In [ ]:
urbanization_by_factors = df.groupby(['Industry_Type'])['Urbanization_Percentage'].sum().reset_index()


In [ ]:
fig = px.line(urbanization_by_factors,
              x='Industry_Type',
              y='Urbanization_Percentage',
              title='Urbanization Percentage by Industry Type')
fig.show()

In [ ]:
df.head()

Industry_Type         Region       Country  Year  Co2_Emissions_MetricTons  \
0   Construction  North America        Brazil  2010                     89.13   
1         Mining         Europe       Germany  2006                    224.81   
2  Manufacturing  South America  South Africa  2017                    179.72   
3   Construction         Europe         India  2018                     23.28   
4   Construction         Africa         China  2013                    124.54   

   Energy_Consumption_TWh  Automobile_Co2_Emissions_MetricTons  \
0                   90.14                                98.37   
1                  931.70                                10.81   
2                  255.14                                55.36   
3                  887.29                                78.99   
4                  923.03                                65.93   

   Industrial_Co2_Emissions_MetricTons  Agriculture_Co2_Emissions_MetricTons  \
0                               118.36                                 31.41   
1                                66.73                                 39.45   
2                               111.67                                  1.25   
3                               123.60                                 46.81   
4                                52.32                                 35.67   

   Domestic_Co2_Emissions_MetricTons  Population_Millions  GDP_Billion_USD  \
0                               0.77               941.06         13096.32   
1                               0.21              1421.94         24337.71   
2                               4.97               522.75         24523.17   
3                              13.77              1304.60         12616.18   
4                              13.91              1437.68          4476.00   

   Urbanization_Percentage  Renewable_Energy_Percentage  \
0                    52.79                         7.78   
1                    50.18                        31.52   
2                    65.22                         5.91   
3                    23.69                         7.52   
4                    94.64                         8.54   

   Industrial_Growth_Percentage  Transport_Growth_Percentage  
0                         11.17                         2.93  
1                         13.34                         9.30  
2                         -9.88                         4.77  
3                         -0.64                         8.21  
4                          5.98                         0.84

#10.Find Out The Renewable_Energy_Percentage

In [ ]:
renewable_energy_by_factors = df.groupby(['Region', 'Country', 'Year'])['Renewable_Energy_Percentage'].sum().reset_index()

fig = px.line(renewable_energy_by_factors,
              x='Year',
              y='Renewable_Energy_Percentage',
              color='Country',
              facet_col='Region',
              title='Renewable Energy Percentage by Region, Country, and Year')

fig.show()

In [ ]:
renewable_energy_by_factors = df.groupby(['Industry_Type'])['Renewable_Energy_Percentage'].sum().reset_index()


In [ ]:
fig = px.scatter(renewable_energy_by_factors,
                 x='Industry_Type',
                 y='Renewable_Energy_Percentage',
                 size='Renewable_Energy_Percentage',
                 color='Industry_Type',
                 hover_name='Industry_Type',
                 title='Renewable Energy Percentage by Industry Type (Bubble Chart)',
                 size_max=30)

fig.show()

#11.Find Out The Industrial_Growth_Percentage

In [ ]:
industrial_growth_by_factors = df.groupby(['Region', 'Country', 'Year'])['Industrial_Growth_Percentage'].sum().reset_index()


In [ ]:
fig = px.area(industrial_growth_by_factors,
             x='Year',
             y='Industrial_Growth_Percentage',
             color='Country',
             facet_col='Region',
             title='Industrial Growth Percentage by Region, Country, and Year')

fig.show()

In [ ]:
industrial_growth_by_factors = df.groupby(['Industry_Type'])['Industrial_Growth_Percentage'].sum().reset_index()


In [ ]:
fig = go.Figure(go.Funnel(
    y=industrial_growth_by_factors['Industry_Type'],
    x=industrial_growth_by_factors['Industrial_Growth_Percentage'],
    textinfo="value+percent initial"))

fig.update_layout(title="Industrial Growth Percentage by Industry Type")
fig.show()

#12.Find Out The Transport_Growth_Percentage.

In [ ]:
transport_growth_by_factors = df.groupby(['Region', 'Country', 'Year'])['Transport_Growth_Percentage'].sum().reset_index()

In [ ]:
fig = px.line(transport_growth_by_factors,
              x='Year',
              y='Transport_Growth_Percentage',
              color='Country',
              facet_col='Region',
              title='Transport Growth Percentage by Region, Country, and Year')

fig.show()

In [ ]:
transport_growth_by_factors = df.groupby(['Industry_Type'])['Transport_Growth_Percentage'].sum().reset_index()

In [ ]:
fig = px.pie(transport_growth_by_factors,
             values='Transport_Growth_Percentage',
             names='Industry_Type',
             title='Transport Growth Percentage by Industry Type (Donut Chart)',
             hole=0.4,
             color_discrete_sequence=px.colors.qualitative.Pastel)

fig.show()

#Conclusion:-

CO₂ emissions by sector reflect the global reliance on carbon-intensive activities and the need for systemic change. The largest contributor to global CO₂ emissions is the energy sector, accounting for around 73% of emissions. This is driven by the burning of fossil fuels for electricity, heat, and transportation. Within energy, electricity and heat production contribute the most, followed by transport.

The transportation sector alone contributes about 16% of global CO₂ emissions, with road transportation being the most significant source due to the extensive use of gasoline and diesel-powered vehicles. This is followed by emissions from aviation and shipping, which are rapidly increasing, especially with the global growth in goods movement and international travel.

The industrial sector contributes roughly 19% of global CO₂ emissions, with cement, steel, and chemical production being the major offenders. These industries are energy-intensive and often rely on fossil fuels for production processes.

Agriculture, forestry, and land use account for approximately 18% of emissions, primarily through deforestation, methane emissions from livestock, and rice paddies.

To reduce global CO₂ emissions, transitioning to renewable energy, improving energy efficiency, and adopting sustainable agricultural practices are critical. Additionally, shifting towards electric vehicles and promoting low-carbon industrial processes can significantly mitigate emissions.